build/oss-fuzz/build.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# Don't need the stdlib and libc++ gives linker error
+CXXFLAGS="${CXXFLAGS/-stdlib=libc++/}"
+
+# Needed for initialize the submodules, the submodule directories are empty.
+# Some header files (proj_api.h here) do not actually exist in cockroach, it exists in the submodule.
+git submodule update --init --recursive
+
+# Install dependencies for compile_native_go_fuzzer
+go install github.com/AdamKorcz/go-118-fuzz-build@latest
+go get github.com/AdamKorcz/go-118-fuzz-build/testing
+
+bazel clean --expunge
+
+# Generate artifacts for building the binary
+bazel run pkg/gen:code
+bazel run pkg/cmd/generate-cgo:generate-cgo --run_under "cd $SRC/cockroach && "
+
+# File is not being built via bazel and outdated (gives compilation errors in SessionIDEncoding)
+rm pkg/sql/pgwire/fuzz.go
+
+# Rename as go-118-fuzz-build cant read _test.go files (used by sessionID and frontier)
+mv ./pkg/util/span/frontier_test.go pkg/util/span/frontier_test_fuzz.go
+
+# Build the native go fuzz targets
+compile_native_go_fuzzer ./pkg/keys FuzzPrettyPrint fuzzPrettyPrint
+
+compile_native_go_fuzzer ./pkg/sql/sqlliveness/slstorage FuzzSessionIDEncoding fuzzSessionIDEncoding
+
+compile_native_go_fuzzer ./pkg/util/span FuzzBtreeFrontier fuzzBtreeFrontier
+
+# hacky fix since we've already copied these files (used in coverage only)
+sed -i '/cp "\${fuzzer_filename}" "\${OUT}\/rawfuzzers\/\${fuzzer}"/d' /usr/local/bin/compile_native_go_fuzzer > /dev/null 2>&1
+compile_native_go_fuzzer ./pkg/util/span FuzzLLRBFrontier fuzzLLRBFrontier
+
+compile_native_go_fuzzer ./pkg/ccl/pgcryptoccl/pgcryptocipherccl FuzzEncryptDecryptAES fuzzEncryptDecryptAES
+
+compile_native_go_fuzzer ./pkg/ccl/pgcryptoccl/pgcryptocipherccl FuzzNoPaddingEncryptDecryptAES fuzzNoPaddingEncryptDecryptAES
+
+compile_native_go_fuzzer ./pkg/storage FuzzEngineKeysInvariants fuzzEngineKeysInvariants
+
+# Build old fuzz targets which used `gofuzz`
+compile_go_fuzzer /src/cockroach/pkg/util/uuid Fuzz fuzzuuid
```

Comment:
Could we do the same with the other `gofuzz` targets including the one above, `pkg/sql/pgwire/fuzz.go`? Or did you run into build issues?

Reply:
Yes we can! `pkg/sql/pgwire/fuzz.go` in particular is just outdated ([a function that it calls now accepts different args](https://github.com/cockroachdb/cockroach/blob/8f72a01e81d4751c6615a9a864b8e32076565b09/pkg/sql/pgwire/fuzz.go#L53)) and throws errors when built.